notebooks/ir_ima_visualization/ima_visualization_and_differencing.py

```diff
@@ -0,0 +1,327 @@
+    '''
+    
+    
+    median_full_frame = np.nanmedian(cube, axis = (0,1))
+    median_lhs = np.nanmedian(cube[lhs_region['y0']:lhs_region['y1'],
+                                      lhs_region['x0']:lhs_region['x1'],:], axis = (0,1))
+    median_rhs = np.nanmedian(cube[rhs_region['y0']:rhs_region['y1'],
+                                      rhs_region['x0']:rhs_region['x1'],:], axis = (0,1))
+    
+    
+    return median_full_frame, median_lhs, median_rhs
+
+def get_median_std_fullframe_lhs_rhs(cube, lhs_region, rhs_region):
+     
+    '''
+    Compute the standard deviation of the median in the full-frame image, the user-defined left side region, 
+    and the user-defined right side region. 
+
+    Parameters
+    ----------
+    cube : array-like
+       1024x1024xNSAMP datacube of the IR image in ascending time order, 
+       where NSAMP is the number of samples taken.
+
+    lhs_region:  dict
+       The four corners (x0, x1, y0, y1) of the left hand region.
+
+    rhs_region : dict
+       The four corners (x0, x1, y0, y1) of the right hand region.
+
+
+    Returns
+    -------
+    standard_dev_fullframe : array of floats
+        The standard deviation of the median signal of the full frame of each read.
+        
+    standard_dev_lhs : array of floats
+        The standard deviation of the median signal of the left side of each read.
+        
+    standard_dev_rhs : array of floats
+        The standard deviation of the median signal of the right side of each read. 
+    '''
+    
+    
+    standard_dev_fullframe = np.nanstd(cube, axis = (0,1))
+    standard_dev_lhs = np.nanstd(cube[lhs_region['y0']:lhs_region['y1'],
+                                      lhs_region['x0']:lhs_region['x1'],:], axis = (0,1))
+    standard_dev_rhs = np.nanstd(cube[rhs_region['y0']:rhs_region['y1'],
+                                      rhs_region['x0']:rhs_region['x1'],:], axis = (0,1))
+    
+    
+    return standard_dev_fullframe, standard_dev_lhs, standard_dev_rhs
+        
+
+def plot_ramp(ima, integ_time, median_diff_fullframe, median_diff_lhs, median_diff_rhs):
+    
+    '''
+    Plots the signal accumulation ramp of an IMA image. Each point is the median signal (in e-/s) of 
+        the difference between subsequent reads. The median signal difference is plotted for the full 
+        frame image, and the left and right sides. 
+    
+    Parameters
+    -----------
+
+    ima: str
+        Name of the IMA file.
+
+    integ_time : array-like
+        Integration times associated with the datacube in ascending order.
+
+    median_diff_full_frame: array-like
+        The median difference in signal between each full-frame read.
+
+    median_diff_lhs: array-like
+        The median difference in signal between the left side of each read.
+
+    median_diff_rhs: array-like
+        The median difference in signal between the right side of each read.
+
+    '''
+    md = median_diff_fullframe
```

Comment:
Better to just keep the original variable names and change the ones in the plotting functions directly.